src/helpers/TripHelper.js

```diff
@@ -40,6 +42,39 @@ class TripHelper {
       return Response.errorMessage(req, res, err.message, 500);
     }
   }
+
+  /**
+   * find the most traveled destination
+   * @static
+   * @memberof class TripChecker
+   * @returns {object} data
+   */
+  static async findMostTraveledDestination() {
+    const destinationObj = {
+      attributes: ['destinationId'], raw: true
+    };
+    const getAllDestinations = await CommonQueries.findAll(trips, destinationObj);
+    const getDestinationIds = getAllDestinations.map(i => i.destinationId);
+    const countHolder = [];
+    const newArray = [];
+
+    for (let i = 0; i < getDestinationIds.length; i++) {
+      let count = 0;
+      for (let z = 0; z < getDestinationIds.length; z++) {
+        if (getDestinationIds[z] === getDestinationIds[i]) count++;
+      }
+      countHolder.push(count);
+      newArray.push({ counter: count, cityId: getDestinationIds[i] });
+    }
+    const maxNumber = Math.max(...countHolder);
+    const frequentCityId = newArray.filter((item) => {
+      if (item.counter === maxNumber) {
+        return { cityId: item.cityId, counts: item.counter };
+      }
+    });
+    const [{ cityId, counter }] = frequentCityId;
+    return { cityId, counter };
+  }
```

Comment:
Wow! I guess you spend a lot of time  to come up with such a beautiful algorithm. Good work.

Reply:
You are right @hezronkimutai, Thank you so much for that acknowledgment!